src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/volume/AzureVolumePluginTest.java

```diff
@@ -0,0 +1,468 @@
+    // call was successful.
+    @Test
+    public void testDoRequestInstanceSuccessfully() throws Exception {
+        // set up
+        String resourceName = AzureTestUtils.RESOURCE_NAME;
+        Creatable<Disk> diskCreatable = Mockito.mock(Creatable.class);
+        Mockito.when(diskCreatable.name()).thenReturn(resourceName);
+        
+        Observable<Indexable> observable = Mockito.mock(Observable.class);
+        PowerMockito.mockStatic(AzureVolumeSDK.class);
+        PowerMockito.doReturn(observable).when(AzureVolumeSDK.class, "buildCreateDiskObservable", Mockito.any(Creatable.class));
+        
+        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
+        Mockito.doNothing().when(this.operation).subscribeCreateDisk(Mockito.eq(observable));
+        Mockito.doNothing().when(this.plugin).updateInstanceAllocation(Mockito.eq(volumeOrder));
+        
+        PowerMockito.mockStatic(AzureGeneralUtil.class);
+        PowerMockito.doReturn(resourceName).when(AzureGeneralUtil.class, "defineInstanceId", Mockito.anyString());
+        
+        // exercise
+        this.plugin.doRequestInstance(volumeOrder, diskCreatable);
+        
+        // verify
+        PowerMockito.verifyStatic(AzureVolumeSDK.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzureVolumeSDK.buildCreateDiskObservable(Mockito.eq(diskCreatable));
+        
+        Mockito.verify(this.operation, Mockito.times(TestUtils.RUN_ONCE)).subscribeCreateDisk(Mockito.eq(observable));
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).updateInstanceAllocation(Mockito.eq(volumeOrder));
+        
+        PowerMockito.verifyStatic(AzureGeneralUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzureGeneralUtil.defineInstanceId(Mockito.eq(resourceName));
+    }
+    
+    // test case: When calling the updateInstanceAllocation method, it must verify
+    // that is call was successful.
+    @Test
+    public void testUpdateInstanceAllocationSuccessfully() {
+        // set up
+        int sizeInGB = TestUtils.DISK_VALUE;
+        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
+        Mockito.when(volumeOrder.getVolumeSize()).thenReturn(sizeInGB);
+
+        // exercise
+        this.plugin.updateInstanceAllocation(volumeOrder);
+
+        // verify
+        Mockito.verify(volumeOrder, Mockito.times(TestUtils.RUN_ONCE)).getVolumeSize();
+        Mockito.verify(volumeOrder, Mockito.times(TestUtils.RUN_ONCE))
+                .setActualAllocation(Mockito.any(VolumeAllocation.class));
+    }
+
+    private VolumeInstance createVolumeInstance() {
+        return new VolumeInstance(
+                AzureTestUtils.RESOURCE_NAME, 
+                AzureStateMapper.SUCCEEDED_STATE, 
+                AzureTestUtils.ORDER_NAME, 
+                TestUtils.DISK_VALUE);
+    }
+    
+    
+    private String createResourceId() {
+        String diskIdFormat = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/disks/%s";
```

Comment:
Could you use the constant already created? Look at AzureResourceIdBuilder class.

Reply:
To use the AzureResourceIdBuilder class constant it would have to make your visibility public. We have to come to a consensus on this if we choose to make public all the constants of this class for testing purposes by other classes.